### PR TITLE
Update the implementation of relationship in adapters

### PR DESCRIPTION
**Pull request by timleslie**

### CODE REVIEW

**Comment by gautamsi** [comment]
it would be helpful to see how this works, RFC doc may be?

File: docs/discussions/new-data-schema.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<!--[meta]
+section: discussions
+title: A New Data Schema For Keystone
+[meta]-->
+
+# A New Data Schema For Keystone
+
+We are excited to announce a **new and improved data schema** for KeystoneJS 🎉
```
Comment (jossmac) [inline docs/discussions/new-data-schema.md:8]:
```suggestion
We are excited to announce a **new and improved data schema** for Keystone 🎉
```

File: docs/discussions/new-data-schema.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+The new data schema simplifies the way your data is stored and will unlock the development of new functionality within Keystone.
+
+** 🚨You will need to make changes to your database to take advantage of the new data schema. Read on for details of how to do this.**
```
Comment (jossmac) [inline docs/discussions/new-data-schema.md:11]:
```suggestion
> **Important:** You will need to make changes to your database to take advantage of the new data schema. Read on for details of how to do this.
```

File: docs/discussions/new-data-schema.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+Unfortunately, this trade off is no longer working in our favour.
+Maintaining the denormalised data is now more complex than generating queries against normalised data.
+We are finding that some reported bugs are difficult to resolve due to the complex nature of resolving denormalised data.
+There are also more sophisticated relationship patterns, such as ordered relationships, which are too difficult to implement in the current schema.
+
+## The Solution
+
+To address these problems at their core we have changed our data schema to be normalised and to eliminate the duplicated data.
+These means that our query generation code has become more complex, but this trade off gains us a number of benefits:
```
Comment (jossmac) [inline docs/discussions/new-data-schema.md:46]:
```suggestion
This means that our query generation code has become more complex, but this trade off gains us a number of benefits:
```

File: docs/discussions/new-data-schema.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+1. Eliminates duplicated data in the data.
```
Comment (jossmac) [inline docs/discussions/new-data-schema.md:48]:
```suggestion
1. Eliminates duplicated data in the database.
```

File: docs/discussions/new-data-schema.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+In order to take advantage of these improvements you will need to make some changes to your database.
+In some instances this will simply involving removing tables or columns which are no longer required.
+In other cases you will need to rename some tables or columns and possibly move data from one table to another.
+
+To assist with this process we have written a [Schema Upgrade Guide](/docs/guides/relationship-migration.md), which will take you through the steps to safely transition your database.
+
+## Summary
+
+The new Keystone data schema will simplify and improve the storage of your crucial system data, and will unlock
```
Comment (jossmac) [inline docs/discussions/new-data-schema.md:81]:
```suggestion
The new Keystone data schema will simplify and improve the storage of your crucial system data, and will unlock ???
```

File: docs/discussions/relationships.md [hunk 6]
```diff
@@ -0,0 +1,249 @@
+<!--[meta]
+section: discussions
+title: Relationships
+[meta]-->
+
+# Relationships
+
+KeystoneJS allows you to model your data as a collection of related `Lists`.
```
Comment (jossmac) [inline docs/discussions/relationships.md:8]:
```suggestion
Keystone allows you to model your data as a collection of related `Lists`.
```

File: docs/guides/relationships.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
+<!--[meta]
+section: guides
+title: Configuring Relationships
+[meta]-->
+
+# Configuring Relationships
+
+KeystoneJS allows you to model your data by declaring [relationships](/docs/discussions/relationships.md) between lists.
```
Comment (jossmac) [inline docs/guides/relationships.md:8]:
```suggestion
Keystone allows you to model your data by declaring [relationships](/docs/discussions/relationships.md) between lists.
```

File: docs/guides/new-schema-cheatsheet.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+One to one relationships in the `before` state had a foreign key column on each table.
+In the `after` state, only one of these is stored.
+Because of the symmetry of the one to one relationship, Keystone makes an arbitrary decision about which column to use.
+
+- Identify the foreign key column which is no longer required, and delete it.
+- In our example above we would delete the `Post.author` column.
+
+#### MongoDB
+
+One to one relationships in the `before` state had a field in each collection
```
Comment (jossmac) [inline docs/guides/new-schema-cheatsheet.md:161]:
```suggestion
One to one relationships in the `before` state had a field in each collection.
```

**Pull request merged**